### PR TITLE
Feature/#949 - Refactor Taipy CLI

### LINKED ISSUE #949: BUG-Inputting wrong kwargs in CLI does not return warnings
I wanted to create a scenario-management template project using the CLI so I typed:
```bash
taipy create --teamplate scenario-management
```
Unfortunately, I wrote a typo, and wrote "teamplate" instead of "template" and I got the wrong template

Taipy's CLI should return warnings when inputting non-existing kwargs

### PR DESCRIPTION
**Pull request by trgiangdo**
Resolves https://github.com/Avaiga/taipy/issues/949

In this PR:
- Refactoring Taipy CLI. Now all _CLI classes inherits from `_AbstractCLI`.
- Except from `_GuiCLI` and `_CoreCLI`, all other CLI with specific arguments will raise an error if the argument is not supported. If their is a typo that similar enough to an argument, there will be a suggestion. Something like:
```
$ taipy migrate --repository-type filesystem --slip-backup 
usage: taipy migrate [-h] [--repository-type REPOSITORY_TYPE [REPOSITORY_TYPE ...]] [--skip-backup] [--restore] [--remove-backup]

options:
  -h, --help            show this help message and exit
  --repository-type REPOSITORY_TYPE [REPOSITORY_TYPE ...]
                        The type of repository to migrate. If filesystem or sql, a path to the database folder/.sqlite file should be
                        informed. In case of mongo host, port, user and password must be informed, if left empty it is assumed
                        default values
  --skip-backup         Skip the backup of entities before migration.
  --restore             Restore the migration of entities from backup folder.
  --remove-backup       Remove the backup of entities. Only use this option if the migration was successful.
[2024-03-27 19:46:48.675][Taipy][ERROR] Unknown arguments: --slip-backup. Did you mean: --skip-backup?
[2024-03-27 19:46:48.675][Taipy][ERROR]  Please refer to the help message above for more information.
```
Here, we also print the help message for the user to modify the command.

### CODE REVIEW

File: tests/templates/test_template_cli.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import pytest
+
+from taipy._entrypoint import _entrypoint
+
+
+def test_create_cli_with_wrong_arguments(caplog):
+    with patch("sys.argv", ["prog", "create", "--teamplaet", "default"]):
+        with pytest.raises(SystemExit):
+            _entrypoint()
+        assert "Unknown arguments: --teamplaet. Did you mean: --template?" in caplog.text
```
Comment (jrobinAV) [inline tests/templates/test_template_cli.py:23]: Can you add a test where the name of the template does not exist?
```
taipy create --template lkjdslklj
```

Comment (jrobinAV) [reply tests/templates/test_template_cli.py:23]: It should raise an error.

**Review by jrobinAV — DISMISSED** [review]

File: tests/templates/test_template_cli.py (hunk 1 above)
Comment (trgiangdo) [reply tests/templates/test_template_cli.py:23]: You're right. Good idea

**Comment by jrobinAV** [comment]
@trgiangdo Before merging it, can you check it does not break enterprise edition?

**Comment by trgiangdo** [comment]
it does break 🤣 there is an opening PR to fix that

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**